src/wallet/wallet.cpp

```diff
@@ -2295,6 +2295,24 @@ void CWallet::SetHDChain(const CHDChain& chain, bool memonly)
     hdChain = chain;
 }
 
+void CWallet::CheckNetworkInfo(std::pair<std::string, std::string> readNetworkInfo)
+{
+    LOCK(cs_wallet);
+    std::pair<string, string> networkInfo(PACKAGE_NAME, networkIdString);
+    if (readNetworkInfo != networkInfo)
+        throw std::runtime_error(
+                strprintf("%s: this wallet is for a different network (%s, %s) than the node is configured for (%s, %s)",
```

Comment:
We should have an RPC test that checks this works.

Reply:
RPC test added in 094ebca